### ISSUE TITLE
org2blog fails with 403 on fetching categories on all xml-rpc operations

### ISSUE THREAD

**Issue by steinarb** [org2blog/org2blog#316]
### Description

Org2blog xml-operations fails with a 403 on getting cateories, when creating a new buffer and when attempting to save a draft. The blog in question is hosted at wordpress.com and using org2blog against it from the same laptop and the same emacs, worked in November 2020. I haven't done any intentional config changes to the blog in between (I'm not a frequent blogger).

### Prerequisites

* [X] Did you study the [README](https://github.com/org2blog/org2blog/blob/master/README.org) to learn how the feature you are reporting a bug is expected to work?
* [X] Did you go through the [When Things Go Wrong](https://github.com/org2blog/org2blog#when-things-go-wrong-or-could-go-even-better) process? It covers the steps to review many possibilies
* [X] Did you review the issue tracker to see if this bug is already
reported or resolved? (https://github.com/org2blog/org2blog/issues/294 seems related, but is closed and what fixed it isn't the issue I'm having)
* [X] Did you consider that what you are describing may be a feature request? If you answered yes, review both the issue tracker and the [FUTURE](https://github.com/org2blog/org2blog/blob/master/FUTURE.org)  file to see if the feature request already exists.

### Environment

- The emacs is the same as in November, the .emacs have changed, but not in the org2blog area 
- Windows 10 Home, Version 2004, OS Build 19041.1083, amd64
- Download and manually install (unzip and use the provided command to get a start menu entry)
- GUI
- What is the value of `org2blog/wp-blog-alist`?
   ```
   (("steinars-blogg" :url "http://steinar.bang.priv.no/xmlrpc.php" :username "you" :password "onlyyouknow"))
   ```
   (username and password replaced, but the original value had the correct values)

Now obtain critical library information by:
1. Hitting `M-x`
2. Typing in `org2blog-version-info`
3. Striking `[enter]`
4. The information appears in the minibuffer then gets added to the `kill-ring`, aka your clipboard.
5. Paste that message below:

```
Org2Blog Runtime: Org2Blog 1.1.11, Emacs 26.3, Org Mode 9.1.9, MetaWeblog 1.1.1, XML-RPC 1.6.12
```

Did you go through the `org2blog-user-report` process? If you did then please paste in the log files here [yes]:

```
http -> Contacting host: steinar.bang.priv.no:80
http -> Marking connection as busy: steinar.bang.priv.no:80 #<process steinar.bang.priv.no>
http -> Request is: 
POST /xmlrpc.php HTTP/1.1
MIME-Version: 1.0
Connection: keep-alive
Extension: Security/Digest Security/SSL
Host: steinar.bang.priv.no
Accept-encoding: gzip
Accept-charset: utf-8;q=1, iso-8859-1;q=0.5
Accept: */*
User-Agent: xml-rpc.el/1.6.12 URL/Emacs Emacs/26.3 (Windows-NT; 32bit; i686-w64-mingw32)
Connection: close
Content-Type: text/xml; charset=utf-8
Content-length: 402

<?xml version="1.0" encoding="UTF-8"?>
<methodCall>
  <methodName>metaWeblog.getCategories</methodName>
  <params>
    <param>
      <value>
        <string>1</string>
      </value>
    </param>
    <param>
      <value>
        <string>xxxx</string>
      </value>
    </param>
    <param>
      <value>
        <string>yyyy</string>
      </value>
    </param>
  </params>
</methodCall>

retrieval -> Spinning in url-retrieve-synchronously: nil (#<buffer  *http steinar.bang.priv.no:80*>)
http -> Calling after change function `url-http-wait-for-headers-change-function' for `#<process steinar.bang.priv.no>'
http -> url-http-wait-for-headers-change-function ( *http steinar.bang.priv.no:80*)
http -> Saw end of headers... ( *http steinar.bang.priv.no:80*)
http -> url-http-parse-response called in ( *http steinar.bang.priv.no:80*)
http -> Got a content-length, being smart about document end.
http -> Calling initial content-length for extra data at end of headers
http -> Marking connection as free: steinar.bang.priv.no:80 #<process steinar.bang.priv.no>
http -> url-http-parse-headers called in ( *http steinar.bang.priv.no:80*)
http -> url-http-parse-response called in ( *http steinar.bang.priv.no:80*)
http -> Parsed HTTP headers: class=3 status=301
http -> Contacting host: steinar.bang.priv.no:443
http -> Marking connection as busy: steinar.bang.priv.no:443 #<process steinar.bang.priv.no>
http -> Request is: 
POST /xmlrpc.php HTTP/1.1
MIME-Version: 1.0
Connection: keep-alive
Extension: Security/Digest Security/SSL
Host: steinar.bang.priv.no
Accept-encoding: gzip
Accept-charset: utf-8;q=1, iso-8859-1;q=0.5
Accept: */*
User-Agent: xml-rpc.el/1.6.12 URL/Emacs Emacs/26.3 (Windows-NT; 32bit; i686-w64-mingw32)
Connection: close
Content-Type: text/xml; charset=utf-8
Content-length: 402

<?xml version="1.0" encoding="UTF-8"?>
<methodCall>
  <methodName>metaWeblog.getCategories</methodName>
  <params>
    <param>
      <value>
        <string>1</string>
      </value>
    </param>
    <param>
      <value>
        <string>xxxx</string>
      </value>
    </param>
    <param>
      <value>
        <string>yyyy</string>
      </value>
    </param>
  </params>
</methodCall>

http -> Finished parsing HTTP headers: nil
http -> Spinning waiting for headers...
retrieval -> Spinning in url-retrieve-synchronously: nil (#<buffer  *http steinar.bang.priv.no:80*>)
retrieval -> Spinning in url-retrieve-synchronously: nil (#<buffer  *http steinar.bang.priv.no:443*-149755>)
http -> Calling after change function `url-http-wait-for-headers-change-function' for `#<process steinar.bang.priv.no>'
http -> url-http-wait-for-headers-change-function ( *http steinar.bang.priv.no:443*-149755)
http -> Saw end of headers... ( *http steinar.bang.priv.no:443*-149755)
http -> url-http-parse-response called in ( *http steinar.bang.priv.no:443*-149755)
http -> Saw chunked encoding.
http -> Calling initial chunked-encoding for extra data at end of headers
http -> Reading chunk 0 (435 653 218)
http -> Saw start of chunk 1 (length=206, start=436
http -> Reading chunk 1 (435 653 218)
http -> Got to the end of chunk #1!
http -> Saw start of chunk 2 (length=0, start=642
http -> Saw end of stream chunk!
http -> Removing terminator of last chunk
http -> Marking connection as free: steinar.bang.priv.no:443 #<process steinar.bang.priv.no>
http -> url-http-parse-headers called in ( *http steinar.bang.priv.no:443*-149755)
http -> url-http-parse-response called in ( *http steinar.bang.priv.no:443*-149755)
http -> Parsed HTTP headers: class=4 status=403
http -> Finished parsing HTTP headers: t
http -> Marking connection as free: steinar.bang.priv.no:443 #<process steinar.bang.priv.no>
http -> Activating callback in buffer ( *http steinar.bang.priv.no:443*-149755): #[128 "\302\303\304p#\210\300\305\240\210\301p\240\207" [(nil) (#<buffer  *http steinar.bang.priv.no:443*-149755>) url-debug retrieval "Synchronous fetching done (%S)" t] 5 "

(fn &rest IGNORED)"] ((:error (error http 403) :redirect "https://steinar.bang.priv.no/xmlrpc.php" :peer (:certificate (:version 3 :serial-number "03:7a:6b:d2:53:29:ae:94:20:b9:51:fd:97:ac:c3:3a:9c:a4" :issuer "C=US,O=Let's Encrypt,CN=R3" :valid-from "2021-06-08" :valid-to "2021-09-06" :subject "CN=tls.automattic.com" :public-key-algorithm "RSA" :certificate-security-level "Medium" :signature-algorithm "RSA-SHA256" :public-key-id "sha1:65:4f:af:0f:0f:3b:34:c0:e8:bc:73:5e:4d:74:94:0b:ba:4c:75:d5" :certificate-id "sha1:09:3c:11:0d:9d:8f:30:cd:7d:6b:50:e3:7b:2b:6b:d7:6e:7e:61:1c") :key-exchange "ECDHE-RSA" :protocol "TLS1.2" :cipher "CHACHA20-POLY1305" :mac "AEAD")))
retrieval -> Synchronous fetching done (#<buffer  *http steinar.bang.priv.no:443*-149755>)
http -> Spinning waiting for headers...
```

### Server

- What is the URL of your blog? https://steinar.bang.priv.no
- Do you have a web browser to use for testing? Yes
- Can you reach that URL using that browser? Yes
- Can you log into your blog using that browser? Yes
- Can you create or edit a post using that browser? Yes

### Hardware Issues

It is easy to diagnose odd computer and router behavior and that might have been reported to address issues working with WordPress's XML-RPC. However for most of us rebooting the server is neither trivial nor easy and in fact can be a very bad idea. Only consider that route if you know what you are doing.

- Have you tried rebooting your computer? Not for this, but I will reboot soon for an upgrade. Don't expect it will fix anything
- Have you tried rebooting your router? Not mine to boot (I'm on vacation).
- Have you tried rebooting your server? The server is hosted by wordpress.com, not mine to boot

### Minimal Reproducible Example

Read about creating a [Minimal Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) to learn about the best way to write a BR. Although the post is tailored for their discussion board, the techniques and strategies apply perfectly to report a bug. After reading about their approach a couple of times, please move forward with the BR.

### Steps to Reproduce

1. M-x  org2blog-user-interface RET
2. 'e' to create a new buffer
3. org2blog asks if it should log in and I say yes
4. org2blog logs in and fails after a while saying that it couldn't get any categories, a new buffer opens

### Expected behavior

That the buffer opens without any error messages and that I would be able to save a draft (and post a blog entry).

### Actual behavior

I get error messages when opening a buffer and can't save drafts

### Additional context

I was originally running org2blog 1.1.1 from melpa stable, and upgraded to 1.1.11 from melpa stable and then upgraded to 20210422.339 from melpa without either fixing the issue.

The log above, is from version 1.1.11.

**Comment by steinarb** [comment]
Rebooted the laptop where emacs runs because of system updates. org2blog still fails after the reboot.

**Comment by grettke** [comment]
Resolved in #308
